contracts/klmd-custom-vesting/src/state.rs

```diff
@@ -95,6 +115,27 @@ impl Account {
 
         Ok(vested_token)
     }
+
+    pub fn prevested_amount(&self, block_info: &BlockInfo) -> StdResult<Uint128> {
+
+        if block_info.time < self.registration_time {
+            return Ok(Uint128::zero());
+        }
+
+        if block_info.time >= self.start_time {
+            return Ok(self.vesting_amount.clone());
+        }
+
+        let prevested_token = self.vesting_amount
+                    .checked_mul(Uint128::from(block_info.time.nanos() - self.registration_time.nanos()))?
+                    .checked_div(Uint128::from(self.start_time.nanos() - self.registration_time.nanos()))?;
+        
+        if self.prevesting_amount > prevested_token {
```

Comment:
@cikleo we need to check that this logic is okay with the requirements. Can you check with Avi?

Reply:
This logic was approved